src/components/Nav/Nav.js

```diff
@@ -1,19 +1,22 @@
 import React from 'react';
 import styles from './Nav.module.css';
+import CartSvg from '../../helper-components/CartSvg';
 
 const Nav = ({
```

Comment:
Indeed this component is a bit messy. What went wrong here (I think) is that you tried to make the component generic with `leftButtons` and `rightButtons` props, but then you added conditional logic which more or less "locks" you to a specific set of buttons.